File: 12/js/1.js
const fs = require("fs");
const { get_start, get_end, create_map, get_shortest_path } = require("./utils");

(function () {
  const contents = fs.readFileSync("../inputSample.txt").toString();
  const map = create_map(contents);

  const path = get_shortest_path(map, get_start(map), get_end(map));
  console.log("Path:", path.map(([x, y]) => `[${x},${y}]`).join(','));
  console.log(`Path length: ${path.length}`);
})();

File: 12/hs/1.hs
import Data.List
import System.IO
import Utils

--- BUG: Exploring to a position with new previous vertex?? 

main = do
  h <- openFile "../input.txt" ReadMode
  contents <- hGetContents h
  let heights = (map . map) get_val $ lines contents
      Just start = get_start heights
      Just end = get_end heights
      path = shortest_path heights start end
  print heights
  -- print $ "Path: " ++ (intercalate "," $ map (\(x, y) -> "(" ++ show x ++ "," ++ show y ++ ")") path)
  -- print $ "Path length: " ++ show (length path) -- Print path length; To get the number of steps, subtract one
  hClose h